iron-fish/website-testnet
# Editable Settings

Changes:

- [x] - 🧹 Removed `API_KEY` usage. It's a hold-over from a former age, and now it is dead.
- [x] - 🏋🏽 Added a new function in `apiClient` for updating a user. It's called (surprisingly) `updateUser`. It takes an object blob and passes it to the API via `PUT`.
- [x] - 🔧 Fixed a several bugs in `useField` which would cause it to infinitely re-render
- [x] - Field updates:
  - Disabled email
  - Graffiti with warning-on-touch
  - Split out Discord + Telegram into separate non-required fields.
  - Country Selector with no default
- [x] Wrote a bunch of smoke tests - but to aid readability, I have moved these changes to https://github.com/iron-fish/website-testnet/pull/164
- [x] - 🦾 Improved the behavior of `hooks/useLogin` so that it can be reloaded
- [x] - Updated `/users/[id]` to support picking tabs via query string: `?tab=settings`

Testing Process:

1. Make sure you can log in (use the leaderboard to verify whether you've already created a user on staging)
2. Go to your user (click your username in the upper right). All of the tabs should still work, but now the settings tab should allow one to actually change values. (If you take a look at the network tab you should see the data verifies this) Change some values for fun / profit!
3. You shouldn't be able to see other people's settings
